File: microservices/product-composite/api/src/main/java/fr/edjaz/api/composite/product/ProductAggregate.java
package fr.edjaz.api.composite.product;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@RequiredArgsConstructor
public class ProductAggregate {
    private final int productId;
    private final String name;
    private final int weight;
    private final List<RecommendationSummary> recommendations;
    private final List<ReviewSummary> reviews;
    private final ServiceAddresses serviceAddresses;

    public ProductAggregate() {
        productId = 0;
        name = null;
        weight = 0;
        recommendations = null;
        reviews = null;
        serviceAddresses = null;
    }

}

File: microservices/product-composite/api/src/main/kotlin/fr/edjaz/api/composite/product/ProductAggregate.kt
package fr.edjaz.api.composite.product

data class ProductAggregate (
  var productId: Int = 0,
  var name: String? = null,
  var weight: Int = 0,
  var recommendations: List<RecommendationSummary>? = null,
  var reviews: List<ReviewSummary>? = null,
  var serviceAddresses: ServiceAddresses? = null
)
